gradio/chat_interface.py

```diff
@@ -579,14 +579,12 @@ def response_as_dict(self, response: MessageDict | Message | str) -> MessageDict
             new_response = response
         return cast(MessageDict, new_response)
 
-    async def _submit_fn(
+    def _format_input(
         self,
         message: str | MultimodalData,
         history_with_input: TupleFormat | list[MessageDict],
-        request: Request,
-        *args,
-    ) -> tuple[TupleFormat, TupleFormat] | tuple[list[MessageDict], list[MessageDict]]:
-        if self.multimodal and isinstance(message, MultimodalData):
+    ) -> tuple[str | dict, TupleFormat | list[MessageDict]]:
+        if isinstance(message, MultimodalData):
             remove_input = (
```

Comment:
nit: could refactor to `remove_input = len(message.files) + int(message.text is not None)`